blocks/init/src/Blocks/custom/featured-posts/components/featured-posts-options.js

```diff
@@ -84,28 +153,70 @@ export const FeaturedPostsOptions = ({ attributes, setAttributes }) => {
 				<Spinner />
 			}
 
-			{(postTypeOptions[0] && posts) ?
+			{(postTypeOptions && taxonomyOptions) ?
 				<SelectControl
-					label={__('Posts Items', 'eightshift-frontend-libs')}
-					value={posts}
-					multiple
-					options={postsOptions}
+					label={sprintf(__('Filter by %s available taxonomies', 'eightshift-frontend-libs'), _.startCase(_.toLower(postType)))}
+					help={sprintf(__('If `No Filter` value is selected your %s posts will not be filtered by any category.', 'eightshift-frontend-libs'), _.startCase(_.toLower(postType)))}
+					value={taxonomy}
+					options={taxonomyOptions}
 					onChange={(value) => {
 						setAttributes({
 							query: {
 								...queryProps,
-								posts: value[0] ? value : [],
+								taxonomy: value,
+								terms: [],
+								posts: [],
 							},
 						});
 					}}
 				/> :
 				<Spinner />
 			}
 
+			{(taxonomyOptions && taxonomy) &&
+				<CustomSelect
+					label={sprintf(__('Filter by %s taxonomy', 'Tvornica'), _.startCase(_.toLower(taxonomy)))}
+					help={sprintf(__('If `No Filter` value is selected your %s posts will not be filtered.', 'Tvornica'), _.startCase(_.toLower(taxonomy)))}
+					options={termsOptions}
+					value={terms}
+					multiple={true}
+					onChange={(value) => {
+						setAttributes({
+							query: {
+								...queryProps,
+								terms: value[0] ? value : [],
+								posts: [],
+							},
+						});
+					}}
+				/>
+			}
+
+			{(postTypeOptions && !taxonomy) &&
+				<CustomSelect
+					label={sprintf(__('Filter by %s articles', 'eightshift-frontend-libs'), _.startCase(_.toLower(postType)))}
+					help={sprintf(__('If `No Filter` value is selected your %s articles will not be filtered.', 'eightshift-frontend-libs'), _.startCase(_.toLower(postType)))}
```

Comment:
Same as the above with extracting to a variable.

```suggestion
					label={sprintf(__('Filter by %s articles', 'eightshift-frontend-libs'), _.startCase(_.toLower(postType)))}
					help={sprintf(__('If the `No Filter` value is selected, your %s articles will not be filtered.', 'eightshift-frontend-libs'), _.startCase(_.toLower(postType)))}
```